### PR TITLE
Make SourceConn clonable and bump Arrow/Datafusion

### PR DESCRIPTION
**Pull request by gruuya**
Hi folks 👋🏻 

We're using connector-x to build a [remote table](https://github.com/splitgraph/seafowl/pull/190) querying capability in [Seafowl](https://seafowl.io/) (an analytical, cache friendly, time travel-able database we're building on top of Datafusion).

Given that on one hand we rely on tokio async runtime, while on the other hand [some of the DB drivers](https://github.com/sfu-db/connector-x/issues/218) block the thread when executing queries, we're forced to run the queries in a closure using [`spawn_blocking`](https://docs.rs/tokio/0.2.22/tokio/task/fn.spawn_blocking.html).

In turn, instead of creating a single `SourceConn` upon instantiating the remote table, we're forced to instantiate a new `SourceConn` for each query execution,  since otherwise we can't pass the ref to the `get_arrow` function inside the closure due to lifetimes incompatibilities (`borrowed data escapes outside of associated function`).

Long story short, to mitigate this we'd really only need `SourceConn` to be cloneable (unless this is for some reason a bad idea?), as then we could get away with keeping a single one in remote table struct and clone it or it's fields when needed, hence this PR.

Additionally, I bumped Arrow and Datafusion to the latest available versions (26 and 14 respectively).

### CODE REVIEW

**Comment by alswang18** [comment]
@wangxiaoying I think this should be fine. We are only making a string container clonable as far as I can tell for this PR. Would love to know your thoughts.

**Comment by wangxiaoying** [comment]
Hi @gruuya , thanks for the PR! To pass the [CI](https://github.com/sfu-db/connector-x/actions/runs/3423127595/jobs/5706315619), I made a minor change to align `arrow` version in rust with cpp and python crates. I'm going to close this PR and redirect to #399 .

**Comment by gruuya** [comment]
Great, thank you!
